src/librustdoc/html/static/js/externs.js

```diff
@@ -8,10 +8,30 @@ function initSearch(searchIndex){}
 
 /**
  * @typedef {{
- *   raw: string,
- *   query: string,
- *   type: string,
- *   id: string,
+ *     isExact: boolean,
+ *     name: string,
+ *     fullPath: Array<string>,
+ *     pathWithoutLast: Array<string>,
+ *     pathLast: string,
```

Comment:
Why do we need all three of these? It seems like we would be fine with `fullPath` or (`pathWithoutLast`, `pathLast`).

Trying to understand these better. For example `std::string::String::from_bytes`. That would have:

```
{
  name: "std::string::String::from_bytes",
  fullPath: ["std", "string", "String", "from_bytes"],
  pathWithoutLast: ["std", "string", "String"],
  pathLast: "from_bytes"
}
```

Is that right? It would be clearer to just store `path: ["std", "string", "String", "from_bytes"]` and calculate the others as needed.

Reply:
Instead of having to make this computation multiple times for each search entry, I prefer that it's done in one place with meaningful names.

Reply:
That represents a tradeoff of complexity for speed. I think the tradeoff is not worth it here. The computation is very small - for instance, pathWithoutLast is just slices off the last element of the array.

Is there any one of these that is accessed so frequently it needs to be memoized?
And `.name` isn't refer

Reply:
> That represents a tradeoff of complexity for speed. I think the tradeoff is not worth it here. The computation is very small - for instance, pathWithoutLast is just slices off the last element of the array.

Having this called on all elements where we were discussing that performance was somewhat sensible seems like an acceptable tradeoff.